### ISSUE TITLE
upload file

### ISSUE THREAD

**Issue by drewads** [drewads/Developer#6]
On click of upload button, client opens a select file navigator. Client sends the selected file and the working directory to the server, which saves the file in the given directory. Server sends the updated state of the working directory to the client.

**Comment by drewads** [comment]
See if we can get this working using the [FormData object](https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects) to send a file input from a form and [Formidable](https://www.npmjs.com/package/formidable) to parse the incoming file. May not need to use formidable at all and could maybe just use the [fs module](https://nodejs.org/api/fs.html).
Some extra documentation on this: [stackOverflow](https://stackoverflow.com/questions/6974684/how-to-send-formdata-objects-with-ajax-requests-in-jquery).

**Comment by drewads** [comment]
In backendTest.html in the test folder, get [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object using [FileList](https://developer.mozilla.org/en-US/docs/Web/API/FileList) and input element. Then, send a save request to the server with the File object in the body. See what happens and make updates as needed.

**Comment by drewads** [comment]
When we make a FormData, we use formData.append('filepath with name of file we want to save', file). See if this works. Alternatively, we could save everything in the same directory, in which case we just add a form field that is dir. Or, we could add as many form fields as there are files.

**Comment by drewads** [comment]
https://stackoverflow.com/questions/6884382/node-js-formidable-upload-with-xhr?rq=1

**Comment by drewads** [comment]
https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop
when we are working with multiple files, we need to await all of the files renaming before returning success. Potentially want to use the promise.all with array.map method of doing this. May also want to keep a list of files that have been moved (maybe just push onto list their filepath when we have moved them). If something goes wrong, we then need to delete those files and the ones that haven't been moved.
